compiler/rustc_target/src/spec/mod.rs

```diff
@@ -2615,6 +2615,18 @@ impl TargetOptions {
             }
         })
     }
+
+    pub(crate) fn has_neg_feature(&self, search_feature: &str) -> bool {
```

Comment:
hm, shouldn't this be in the `target_features` module?

Reply:
I feel like this is more just a getter for the target spec, and all the target spec functionality is in this file.

Reply:
Okay. As long as it's here, can we leave a comment that we should probably try to simply define features so that they are off-by-default, and then enable other features, instead of disabling them? (it has weird compositional effects with `-Ctarget-cpu`)

Reply:
It's not really our choice though? If LLVM makes a feature on-by-default we just have to live with that.

Reply:
```suggestion
    /// We should try to minimize the number of negative features defined for any target
    /// because they often interact very poorly with how `-Ctarget-cpu` works.
    /// However, their mere existence is still a possibility, so!
    pub(crate) fn has_neg_feature(&self, search_feature: &str) -> bool {
```

Reply:
@RalfJung It's more about avoiding using `-feature` when unnecessary.

Reply:
Ah I see, makes sense.